### DISCUSSION TITLE
Quering json blob

### DISCUSSION THREAD

**Discussion by woutercouvaras** [pocketbase/pocketbase#4512] [Q&A] [answered]
Hi!

I'm loving PB.  It's such well thought out software!  I've been able to build out an auth/profile management flow in the shortest time...so amazing! 

I'm now trying something a bit different.  I'm trying to query a blob of json in a json column.  If I run the query directly in the sqlite terminal, it returns what I expect, but when I try to run it in PB, I don get an empty result (I'm running this in a custom endpoint).

Here's the json blob in the db:

```
{
  "hero": {
    "_key": "nx8SLx6Ke",
    "preHeading": "Call of",
    "heading": "The Wild",
    "gallery": [
      {
        "_key": "SOajU7pvu",
        "image": [
          {
            "__type": "some type",
            "className": "Class Name",
            "id": "Czh7ocuWxR"
          }
        ]
      }
    ]
  }
}
```

My PB endpoint:

```
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.GET("/api/json-query", func(c echo.Context) error {
			type Doc struct {
				Id      string        `db:"id" json:"id"`
				Content types.JsonMap `db:"content" json:"content"`
			}

			docs := []Doc{}

			err := app.Dao().DB().
				NewQuery("SELECT json_extract(content, '$.hero.gallery[0]') AS image FROM doc WHERE content->>'$.hero._key' = 'nx8SLx6Ke'").
				All(&docs)

			if err != nil {
				return err
			}

			log.Default().Printf("docs: %v", docs)

			return c.JSON(http.StatusOK, map[string]interface{}{"docs": docs})
		} /* optional middlewares */)
		return nil
	})
```

If I print the result, you can see the empty map:

```
docs: [{ map[]}]
```

And the api obv corresponds:

```
{
	"docs": [
		{
			"id": "",
			"content": {}
		}
	]
}
```

My sqlite terminal output:
```
image
{"_key":"SOajU7pvu","image":[{"__type":"some type","className":"Class Name","id":"Czh7ocuWxR"}]}
```

Perhaps it's because these types of queries are not supported, but I'm really hoping it's just me being stupid :D

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
> If I run the query directly in the sqlite terminal, it returns what I expect, but when I try to run it in PB, I don get an empty result 

It returns empty result because the fields you are selecting (in the above example "image") doesn't exist in the `Doc` struct you are scanning the result into.

**Reply by woutercouvaras** [reply]
Wow, you replied so fast!!  Thank you!

As I said, I was hoping it's me that's done something stupid, and so it was :D

I'm new to go and types, so I've clearly got some more learning to do, esp returning dynamic results for queries similar to this, but you're answer is so helpful, thanks again!

I'm loving what you're doing with PB ❤️
